assembly/src/assembler/context.rs

```diff
@@ -250,11 +250,17 @@ impl AssemblyContext {
     // HELPER METHODS
     // --------------------------------------------------------------------------------------------
 
-    /// Returns the context of the procedure currently being complied, or None if module or
+    /// Returns the context of the procedure currently being compiled, or None if module or
     /// procedure stacks are empty.
     fn current_proc_context(&self) -> Option<&ProcedureContext> {
         self.module_stack.last().and_then(|m| m.proc_stack.last())
     }
+
+    /// Returns the name of the procedure currently being compilied, or None if module or
+    /// procedure stacks are empty.
+    pub(crate) fn current_proc_context_name(&self) -> Option<&str> {
+        self.current_proc_context().map(|p| p.name().as_ref())
```

Comment:
shouldn't the translation to `#main` happen here?

Reply:
I'm not sure I understand. Here we just fetch the string from proc context, delegating any logic to its implementation. It might return `#main` if we are on the base procedure, or some module path with the instruction.

Reply:
humm, I assume it return `None` when inside main, because later on the code there is an `unwrap_or("#main")`:

https://github.com/0xPolygonMiden/miden-vm/pull/756/files#diff-a50e9d9e79218b30b038742cf08858ad7f34917a9fb1f0a66550f1d11fc85e43R106

If this could have `#main` already, then why do we need that unwrap?

Reply:
Hmm that's something to think. We follow the same strategy of `current_proc_context`, that is to return an Option on whether or not we are inside of a procedure.

For `main`, we are not. But then it would be good to integrate the logic. I can rename the function to `current_procedure_name`, I'm just unsure if it's still a procedure or something else (maybe a function?)

Reply:
Since `main` isn't a procedure, maybe renaming this to `current_context_name` would be clearer, and then you could do the translation to `#main` here

Reply:
Makes sense,renamed to `current_context_name` and behaving as @hackaugusto suggested!